docs(storage): adds a description of tiered storage (#9866)
strimzi/strimzi-kafka-operator: documentation/modules/configuring/ref-storage-tiered.adoc

[id='ref-tiered-storage-{context}']
= Tiered storage (preview)

[role="_abstract"]
Tiered storage introduces a flexible approach to managing Kafka data whereby log segments are moved to a separate storage system. 

> For example, you can combine the use of local disks on brokers for frequently accessed data and offload older or less frequently accessed data from the local disks to more cost-effective, scalable remote storage solutions, such as Amazon S3, without compromising data accessibility and durability.

Reviewer: ```suggestion
For example, you can combine the use of block storage on brokers for frequently accessed data and offload older or less frequently accessed data from the block storage to more cost-effective, scalable remote storage solutions, such as Amazon S3, without compromising data accessibility and durability.
```

WARNING: Tiered storage is an early access Kafka feature, which is also available in Strimzi. 
Due to its https://kafka.apache.org/documentation/#tiered_storage_limitation[current limitations^], it is not recommended for production environments. 

Tiered storage requires an implementation of Kafka's `RemoteStorageManager` interface to handle communication between Kafka and the remote storage system, which is enabled through configuration of the `Kafka` resource.

> Strimzi uses Kafka's `TopicBasedRemoteLogMetadataManager` for Remote Log Metadata Management (RLMM) when custom tiered storage is enabled.

Reviewer 2: I would add this link https://github.com/apache/kafka/blob/trunk/storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManager.java as it is already in some previous content.

[id='ref-tiered-storage-{context}']
= Tiered storage (preview)

[role="_abstract"]
Tiered storage introduces a flexible approach to managing Kafka data whereby log segments are moved to a separate storage system. 

> For example, you can have fast local disks on brokers and offload older or less frequently accessed data from the local disks to more cost-effective, scalable remote storage solutions, such as Amazon S3, without compromising data accessibility and durability.

Reviewer: ```suggestion
For example, you can combine the use of regular disks on brokers for frequently accessed data and offload older or less frequently accessed data from the local disks to more cost-effective, scalable remote storage solutions, such as Amazon S3, without compromising data accessibility and durability.
```

> NOTE: Tiered storage is available a technology preview with Strimzi, so it is not suitable for production environments.

Reviewer: Again -> no tech preview, early access in Apache Kafka.

 
Tiered storage requires an implementation of Kafka's `RemoteStorageManager` interface, which is enabled through configuration of the `Kafka` resource.
Strimzi uses Kafka's `TopicBasedRemoteLogMetadataManager` for Remote Log Metadata Management (RLMM) when custom tiered storage is enabled.
The RLMM manages the metadata related to remote storage.  

To use custom tiered storage, do the following:

* Include a tiered storage plugin for Kafka in the Strimzi image by building a custom container image. 
The plugin must provide the necessary functionality for a Kafka cluster managed by Strimzi to interact with the tiered storage solution.

> * Create a custom implementation of Kafka's `RemoteStorageManager` interface for your chosen tiered storage solution. 

Reviewer 3: This sounds like you need to both include a plugin and implement a custom RemoteStorageManager. The plugin is the RemoteStorageManager implementation.

Reviewer: Yeah, I would remover this point. If you are implementing custom tiered storage plugin, you don't need this explanation for that.
